library/core/src/main/java/com/google/android/exoplayer2/video/MediaCodecVideoRenderer.java

```diff
@@ -1071,11 +1071,13 @@ private static boolean deviceNeedsAutoFrcWorkaround() {
    * If true is returned then we fall back to releasing and re-instantiating the codec instead.
    */
   private static boolean codecNeedsSetOutputSurfaceWorkaround(String name) {
-    // Work around https://github.com/google/ExoPlayer/issues/3236 and
-    // https://github.com/google/ExoPlayer/issues/3355.
+    // Work around https://github.com/google/ExoPlayer/issues/3236,
+    // https://github.com/google/ExoPlayer/issues/3355 and
+    // https://github.com/google/ExoPlayer/issues/3439.
     return (("deb".equals(Util.DEVICE) || "flo".equals(Util.DEVICE))
         && "OMX.qcom.video.decoder.avc".equals(name))
-        || ("tcl_eu".equals(Util.DEVICE) && "OMX.MTK.VIDEO.DECODER.AVC".equals(name));
+        || (("tcl_eu".equals(Util.DEVICE) || Util.MODEL.startsWith("BRAVIA"))
```

Comment:
`Util.MODEL.startsWith("BRAVIA")` feels like it will blacklist all Sony TVs for as long as they stick with the BRAVIA branding. Can you come up with something a little more precise, or apply a maximum API level at which the workaround will be applied so that future BRAVIA devices don't automatically get the workaround too?

Thanks!

Reply:
@ojw28 
thanks for the feedback! I've updated BRAVIA workaround to use device name instead of models.